presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftMetastoreClientFactory.java

```diff
@@ -50,14 +72,141 @@ public ThriftMetastoreClientFactory(
     }
 
     @Inject
-    public ThriftMetastoreClientFactory(ThriftMetastoreConfig config, HiveMetastoreAuthentication metastoreAuthentication, NodeManager nodeManager)
+    public ThriftMetastoreClientFactory(
+            ThriftMetastoreConfig config,
+            HiveMetastoreAuthentication metastoreAuthentication,
+            NodeManager nodeManager)
     {
-        this(Optional.empty(), Optional.ofNullable(config.getSocksProxy()), config.getMetastoreTimeout(), metastoreAuthentication, nodeManager.getCurrentNode().getHost());
+        this(
+                buildSslContext(
+                        config.isTlsEnabled(),
+                        Optional.ofNullable(config.getKeystorePath()),
+                        Optional.ofNullable(config.getKeystorePassword()),
+                        Optional.ofNullable(config.getTruststorePath()),
+                        Optional.ofNullable(config.getTruststorePassword())),
+                Optional.ofNullable(config.getSocksProxy()),
+                config.getMetastoreTimeout(),
+                metastoreAuthentication,
+                nodeManager.getCurrentNode().getHost());
     }
 
     public ThriftMetastoreClient create(HostAndPort address, Optional<String> delegationToken)
             throws TTransportException
     {
-        return new ThriftHiveMetastoreClient(Transport.create(address, sslContext, socksProxy, timeoutMillis, metastoreAuthentication, delegationToken), hostname);
+        return new ThriftHiveMetastoreClient(
+            Transport.create(address, sslContext, socksProxy, timeoutMillis, metastoreAuthentication, delegationToken),
+            hostname);
+    }
+
+    private static Optional<SSLContext> buildSslContext(
+            boolean tlsEnabled,
+            Optional<File> keyStorePath,
+            Optional<String> keyStorePassword,
+            Optional<File> trustStorePath,
+            Optional<String> trustStorePassword)
+    {
+        if (!tlsEnabled || (!keyStorePath.isPresent() && !trustStorePath.isPresent())) {
```

Comment:
Should the existence of the wrapped `File` values in the `Optional<File>` parameters be checked here?

Reply:
Those are validated by `try (InputStream in = new FileInputStream(keyStorePath.get())) {}` and `try (InputStream in = new FileInputStream(trustStorePath)) {}`